src/client/src/apps/SpotlightRoute/Spotlight.tsx

```diff
@@ -62,6 +69,37 @@ export const Spotlight: FC = () => {
   const serviceStub = useServiceStub()
   const platform = usePlatform()
 
+  useEffect(() => {
+    if (!contacting) {
+      return;
+    }
+    const subscription = serviceStub
+      .createRequestResponseOperation<DetectIntentResponse[], string>(
+        'nlp',
+        'getNlpIntent',
+        request,
+      )
+      .pipe(
+        tap(() => console.info(JSON.stringify(request))),
+        timeout(10000),
+        take(1),
+      )
+      .subscribe(
+        response => {
+          dispatch({type: 'RECEIVE_RESPONSE', response: response[0]})
+        },
+        (err: any) => {
+          console.error(err)
+          dispatch({type: 'RECEIVE_RESPONSE', response: null})
+        },
+      )
+
+    return () => {
+      subscription.unsubscribe()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why did you need this comment? was it because you didn't want to add `request` as dep?

Reply:
because request keeps changing as you type. And I only want to trigger the request when contacting goes to `true`

Reply:
because we have this 
```
const handleChange: ChangeEventHandler<HTMLInputElement> = e => {
    dispatch({type: 'SET_REQUEST', request: e.target.value})
  }
```

Reply:
that's interesting, I first thought that then `request` should probablt go into ref (not the state), but for now maybe it would be worth just adding a comment on line 99 :

>> because request keeps changing as you type. And I only want to trigger the call when contacting goes to true